src/components/Filter/Filter.tsx

```diff
@@ -1,22 +1,30 @@
 import { DropDownContainer, DropDownHeader, DropDownListContainer, ListItems } from "./Style";
 import Icon from '../Icon/Icon';
-import {useState} from 'react';
+import React,{useState, Dispatch} from 'react';
 import DropIcon from '../../assets/dropdown.svg';
 import DateIcon from '../../assets/date.svg';
+import { FILTER_OPTIONS } from '../../FiltersData';
+
 
 export interface FilterProps {
     filterText: string;
     listItems: string[];
     date?: boolean;
+    setFilterState?: Dispatch<React.SetStateAction<FILTER_OPTIONS>>
 }
 
-const Filter = ({filterText, listItems, date=false}: FilterProps) => {
+const Filter = ({filterText, listItems, date=false, setFilterState}: FilterProps) => {
     const [open, setOpen] = useState(false);
     let icon =  DropIcon;
     if(date){
         icon = DateIcon;
     }
 
+    const onClickHandle = (item: string) => {
+        setFilterState &&
+        setFilterState(item === "Top Headlines" ? FILTER_OPTIONS.TOP : FILTER_OPTIONS.EVERYTHING );
```

Comment:
what will happen if you use this component with another list of items?
that is not "top headlines" or "eveerything", I think it wont work